packages/shared/src/components/PostOptionsMenu.tsx

```diff
@@ -92,9 +93,17 @@ export default function PostOptionsMenu({
       await undo?.();
       client.invalidateQueries(generateQueryKey(feedName, user));
     };
-    displayToast(message, { subject: ToastSubject.Feed, onUndo });
+    displayToast(message, {
+      subject: ToastSubject.Feed,
+      onUndo: undo !== null ? onUndo : null,
+    });
     onRemovePost?.(_postIndex);
   };
+  const { onConfirmDeletePost } = usePostMenuActions({
+    post,
+    onPostDeleted: () =>
+      showMessageAndRemovePost('The post has been deleted', postIndex, null),
```

Comment:
The user can be on the post page when doing this where it will not need to refresh the cache since when the user moves to the feed, it will then fetch on load.